sift/client.py

```diff
@@ -708,7 +725,128 @@ def apply_content_decision(self, user_id, content_id, properties, timeout=None):
+                         'Accept': '*/*',
+                         'User-Agent': self._user_agent()},
+                timeout=timeout))
+
+        except requests.exceptions.RequestException as e:
+            raise ApiException(str(e), url)
+
+    def check(self, properties, timeout=None,  version=None):
+        """The check call is used for sending the OTP provided by the end user to Sift. 
+        Sift then compares the OTP, checks rate limits and responds with a decision whether the user should be able to proceed or not.
+        This call is blocking.  Check out https://sift.com/developers/docs/curl/verification-api/check
+        for more information on our check response structure.
+
+        Args:
+
+            properties:
+                $user_id: id of user
+                $code: The code the user sent to the customer for validation..
+                $verified_event(optional): This will be the event type that triggered the verification. 
+                $verified_entity_id(optional): The ID of the entity impacted by the event being verified.
+
+            timeout(optional): Use a custom timeout (in seconds) for this call.
+            version(optional): Use a different version of the Sift Science API for this call.
+
+        Returns:
+            A sift.client.Response object if the check call succeeded, or raises
+            an ApiException.
+        """
+        if timeout is None:
+            timeout = self.timeout
+
+        self._validate_check_request(properties)
+
+        url = self._check_url()
+
+        try:
+            return Response(self.session.post(
+                url,
+                data=json.dumps(properties),
+                auth=requests.auth.HTTPBasicAuth(self.api_key, ''),
+                headers={'Content-type': 'application/json',
+                         'Accept': '*/*',
+                         'User-Agent': self._user_agent()},
+                timeout=timeout))
+
+        except requests.exceptions.RequestException as e:
+            raise ApiException(str(e), url)
+
+    def _validate_check_request(self, properties):
+        """ This method is used to validate arguments passed to the check method. """
+
+        if not isinstance(properties, dict):
+            raise TypeError("properties must be a dict")
+        elif not properties:
+            raise ValueError("properties dictionary may not be empty")
+
+        user_id = properties.get('$user_id')
+        _assert_non_empty_unicode(user_id, 'user_id', error_cls=ValueError)
+
+        otp_code = properties.get('$code')
+        if otp_code is None:
+            raise ValueError("Code is required")
+
+        if otp_code and type(otp_code) != int:
+            raise ValueError("Code should be  a number")
+
+        verified_event = properties.get('$verified_event')
+        if (verified_event and verified_event not in VERIFIED_EVENTS):
+            raise ValueError(" 'verified_event' must be one of [{0}]".format(
```

Comment:
Do we need this check? I think the server returns an error if the event type is wrong.

You have an extra space at start of the ValueError message.